File: leetcode/problems/language-specific/go/217.contains-duplicate.go
/*
 * @lc app=leetcode id=217 lang=golang
 *
 * [217] Contains Duplicate
 */

// @lc code=start
func containsDuplicate(nums []int) bool {
	if len(nums) <= 1 {
		return false
	}

	mappings := map[int]int{}
	for _, v := range nums {
		mappings[v]++
		if mappings[v] > 1 {
			return true
		}
	}

	return false
}

// @lc code=end

File: leetcode/problems/sessions/2025-10-25_2025-??-??/217.contains-duplicate.py
#
# @lc app=leetcode id=217 lang=python3
#
# [217] Contains Duplicate
#

# @lc code=start

# * Hash Table Solution | O(n) Time | O(n) Space
# * n -> The length of nums array


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        num_idx_map = {}
        for idx, num in enumerate(nums):
            if num in num_idx_map:
                return True

            num_idx_map[num] = idx

        return False


# @lc code=end
